drivers/bridge/bridge.go

```diff
@@ -293,64 +293,46 @@ func (n *bridgeNetwork) getEndpoint(eid string) (*bridgeEndpoint, error) {
 
 // Install/Removes the iptables rules needed to isolate this network
 // from each of the other networks
-func (n *bridgeNetwork) isolateNetwork(others []*bridgeNetwork, enable bool) error {
-	n.Lock()
-	thisConfig := n.config
-	n.Unlock()
-
-	if thisConfig.Internal {
+func (n *bridgeNetwork) isolateNetwork(other *bridgeNetwork, enable bool) error {
+	if n.config.Internal {
 		return nil
 	}
 
-	// Install the rules to isolate this networks against each of the other networks
-	for _, o := range others {
-		o.Lock()
-		otherConfig := o.config
-		o.Unlock()
+	// Install the rules to isolate this networks from the one passed as an argument
+	other.Lock()
+	otherConfig := other.config
```

Comment:
here we are only locking when we copy the `config` pointer, not the config object, is that expected ?

and also I believe we could use a `RLock` since we are not modifying anything